### PR TITLE
Fetch the comments from the datastore and display them in the webpage.

### PR DESCRIPTION
**Pull request by jiaxi312**
- Modify the doGet method to fetch the comments stored in the datastore.

- Display the comments when the webpage loads.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -17,24 +17,38 @@
...
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String json = getInformationInJson();
+        // Fetch the comment from the datastore
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: How about make it private final or private static final?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -17,24 +17,38 @@
...
+        // Store the contents of those comments in a list
+        List<String> commentList = new ArrayList<>();
+        for (Entity entity : comments.asIterable()) {
+            commentList.add((String) entity.getProperty("content"));
+        }
+
+        // Respond the comments as json form
         response.setContentType("application/json");
-        response.getWriter().println(json);
+        response.getWriter().println(new Gson().toJson(commentList));
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: seems like you don't need getInformationInJson?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -73,24 +85,7 @@ private String getParameterWithDefault(HttpServletRequest request, String name,
...
-
-        // Get the current time and adds to the list
-        information.add(new Date().toString());
-
-        // Convert it to json format and returns that json
-        return new Gson().toJson(information);
-    }
-
     private static final String ANONYMOUS = "Anonymous";
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: If you use such initialization, why not make it static?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -17,24 +17,37 @@
...
+        // Store the contents of those comments in a list
+        List<String> commentList = new ArrayList<>();
+        for (Entity entity : comments.asIterable()) {
+            commentList.add((String) entity.getProperty("content"));
+        }
+
+        // Respond the comments as json form
         response.setContentType("application/json");
-        response.getWriter().println(json);
+        response.getWriter().println(new Gson().toJson(commentList));
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Actually, I would suggestion make Gson static too.
Maybe not need to make class scope static, function scope static is enough.

Comment (jiaxi312) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Sorry, I don't quit understand. Do you mean:
`static Gson gson = new Gson();`
`response.getWriter().println(gson.toJson(commentList));` 
But, will it be redundant, cause the gson is only used once here in the whole class?

Comment (zixiyan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You could write
```
static final Gson GSON = new Gson();
response.getWriter().println(GSON.toJson(commentList));
```

used once in code.
But basically it means that you need to new one Gson object for every Get request.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -73,24 +85,7 @@ private String getParameterWithDefault(HttpServletRequest request, String name,
...
-
-        // Get the current time and adds to the list
-        information.add(new Date().toString());
-
-        // Convert it to json format and returns that json
-        return new Gson().toJson(information);
-    }
-
     private static final String ANONYMOUS = "Anonymous";
+    private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: static variable name should be ```DATASTORE```
And I would suggest to rename it to
```
DEFAULT_DATASTORE_SERVICE
```

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -17,24 +17,37 @@
...
-import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String json = getInformationInJson();
+        // Fetch the comment from the datastore
```
Comment (zixiyan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: seems like you're using 4 spaces indent.
per https://google.github.io/styleguide/javaguide.html#s4.2-block-indentation, it seems that you should use 2 spaces indent.

**Review by zixiyan — APPROVED** [review]

**Pull request merged**